content/docs/setup/terraform/_index.md

```diff
@@ -6,52 +6,274 @@ weight = 30
+
+```
+$ chmod +x ./generate-envrc.sh
+$ chmod +x ./setup.sh
+$ chmod +x deploy/tls/*.sh
+```
+
+You have to generate the `envrc`, a file that contains input
+variable that will be used during the `setup` phase. The `envrc` gives us the
+opportunity to create an idempotent workflow and for you is a way to configure
+the `setup.sh` script. For example changing the [osie](/docs/services/osie) version.
+
+```
+$ ./generate-envrc.sh eno1 > envrc
+```
+
+`eno1` is the name of the network interface shared between provisioner and workers.
+Everything is ready and we can run the `setup.sh` script. It changed based on
+the `device_type` in use. There is not an easy way to spot the right one, but
+for the purpose of this documentation it usually is the one with the higher
+index. In this case we have `eno0` and `eno1` and the last one is the right
+network interface.
+
+```
+$ ./setup.sh
+```
+
+This script will use the `envrc` to boot:
+
+* [tink-server](/docs/services/tink)
+* [hegel](/docs/services/hegel)
+* [boots](/docs/services/boots)
+* postgres
+* nginx to serve [osie](/docs/services/osie)
+* A docker registry.
+
+All those daemons will run with docker-compose and you can find the definition
+of it in `tink/deploy/docker-compose.yaml`. Let's start them all:
+
+```
+$ cd ./deploy
+$ source ../envrc
+$ docker-compose up -d
+```
+
+In order to check if all the services are up and running you can use
+docker-compose as well, the output should look similar to this one:
+
+```
+$ docker-compose ps
+        Name                      Command               State                         Ports
+------------------------------------------------------------------------------------------------------------------
+deploy_boots_1         /boots -dhcp-addr 0.0.0.0: ...   Up
+deploy_db_1            docker-entrypoint.sh postgres    Up      0.0.0.0:5432->5432/tcp
+deploy_hegel_1         cmd/hegel                        Up
+deploy_nginx_1         /docker-entrypoint.sh ngin ...   Up      192.168.1.2:80->80/tcp
+deploy_registry_1      /entrypoint.sh /etc/docker ...   Up
+deploy_tink-cli_1      /bin/sh -c sleep infinity        Up
+deploy_tink-server_1   tink-server                      Up      0.0.0.0:42113->42113/tcp, 0.0.0.0:42114->42114/tcp
+```
+
+The workers do not have access to internet, to run the Example Workflow
+successfully we have to proxy a docker image called `hello-world` from Docker
+Hub to the internal registry.
+
+```
+$ docker pull hello-world
+$ docker tag hello-world 192.168.1.1/hello-world
+$ docker push 192.168.1.1/hello-world
+```
+
+## Register the worker
+
+As part of the `terraform apply` output we get the MAC address for the worker.
+It is now time to register it to Tinkerbell. All the following commands have to
+run in the provisioner itself.
+
+{{% notice note %}}
+Replace `1c:34:da:5c:36:88` with the output from `terraform apply`
+{{% /notice %}}
+
+```
+cat > hardware-data.json <<EOF
```

Comment:
In the latest terraform commit, this file already exists.  We can change this instruction to a simple `cat` so they can see the contents of the file.

Reply:
I think I did it, or I thought about doing it. Let me check again

Reply:
I think this is not valid anymore.. The lastest version has your change included